### DISCUSSION TITLE
Switch to Java 11

### DISCUSSION THREAD

**Discussion by p-bakker** [mozilla/rhino#1169] [Ideas]
From @gbrail (see https://github.com/mozilla/rhino/discussions/972#discussioncomment-1915794):

> This is a long long discussion, but I'm ready to say that Rhino can move beyond Java 8 after 1.7.14. Later Java versions are getting pretty ubiquitous, and the log4j thing helped push some people along.
> 
> I'd like to propose Java 11 as the new baseline going forward, with support for newer versions of course.
> 
> Does that make sense?

**Comment by p-bakker** [comment]
To me it does, but a couple of things I'm not sure about:
- What would that do to support for Rhino on Android?
- Is this in any way related to[ (the discussion about) Modular Java](https://github.com/mozilla/rhino/issues/1092)
- Assuming we'll have a v1.7.15 before a v2.0.0: should we switch to Java 11 for v1.7.15 or postpone that for v2.0.0?

**Comment by p-bakker** [comment]
We also have a case for switching to another Java version: #1075

**Comment by p-bakker** [comment]
And I have the idea these are also somewhat related to this topic:
- [Switch to a multiproject/modular source layout](#1092)  
- [Current VMBridge is not suitable for modular JDKs](#1076)
- [Remove RhinoSecurityManager and refactor relevant security-related due to pending deprecation and removal of the Java Security Manager](#1053)

**Comment by gbrail** [comment]
I opened a PR that upgrades the build to require Java 11 minimum, and upgrades some tools so that they will work better with newer versions. If a few of you can take a look, I'm happy to merge that and then start removing some of the older support for older Java versions.

**Comment by gbrail** [comment]
Any more opinions on this? I am ready to merge that PR and at least start the process of moving to Java 11 as a minimum.

**Reply by p-bakker** [reply]
If nobody else objects, I suggest going for it

**Comment by gbrail** [comment]
https://github.com/mozilla/rhino/pull/1344

**Comment by p-bakker** [comment]
Also see https://github.com/mozilla/rhino/discussions/1363

**Comment by p-bakker** [comment]
Discussion closed, as of today Rhino requires Java 11 or higher
